service/interfaces.go

```diff
@@ -0,0 +1,41 @@
+package service
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/dp-image-api/config"
+)
+
+//go:generate moq -out mock/initialiser.go -pkg mock . Initialiser
+//go:generate moq -out mock/mongo.go -pkg mock . IMongo
+//go:generate moq -out mock/server.go -pkg mock . IServer
+//go:generate moq -out mock/healthCheck.go -pkg mock . IHealthCheck
+
+// Initialiser defines the methods to initialise external services
+type Initialiser interface {
+	DoGetHTTPServer(bindAddr string, router http.Handler) IServer
+	DoGetMongoDB(ctx context.Context, cfg *config.Config) (IMongo, error)
+	DoGetHealthCheck(cfg *config.Config, buildTime, gitCommit, version string) (IHealthCheck, error)
+}
+
+// IMongo defines the required methods from MongoDB
+type IMongo interface {
```

Comment:
In go, interfaces should ideally end "-er". Prefixing with I is more common in other languages. I would go with `MongoServer`, `HTTPServer` and `HealthChecker` personally.